From: Jan Kara <jack@suse.cz>
Subject: xfs: Fix kABI due to change in xfs_buf
References: bnc#815356
Patch-mainline: Never

Patch patches.fixes/xfs-fix-race-while-discarding-buffers-V4.patch changes
struct xfs_buf. Since it is exported solely for xfs dmapi modules, just silence
the kabi checker.

Signed-off-by: Jan Kara <jack@suse.cz>

diff -rupX /crypted/home/jack/.kerndiffexclude linux-3.0-SLE11-SP3/fs/xfs/linux-2.6/xfs_buf.h linux-3.0-SLE11-SP3-xfs_kabi/fs/xfs/linux-2.6/xfs_buf.h
--- linux-3.0-SLE11-SP3/fs/xfs/linux-2.6/xfs_buf.h	2013-05-28 10:59:50.074436553 +0200
+++ linux-3.0-SLE11-SP3-xfs_kabi/fs/xfs/linux-2.6/xfs_buf.h	2013-05-28 12:02:33.203238316 +0200
@@ -146,7 +146,9 @@ typedef struct xfs_buf {
 	 * bt_lru_lock and not by b_sema
 	 */
 	struct list_head	b_lru;		/* lru list */
+#ifndef __GENKSYMS__	/* Nobody outside xfs should use this structure */
 	xfs_buf_flags_t		b_lru_flags;	/* internal lru status flags */
+#endif
 	wait_queue_head_t	b_waiters;	/* unpin waiters */
 	struct list_head	b_list;
 	struct xfs_perag	*b_pag;		/* contains rbtree root */
